kedro/utils.py

```diff
@@ -26,3 +27,64 @@ def load_obj(obj_path: str, default_obj_path: str = "") -> Any:
     if not hasattr(module_obj, obj_name):
         raise AttributeError(f"Object '{obj_name}' cannot be loaded from '{obj_path}'.")
     return getattr(module_obj, obj_name)
+
+
+class DeprecatedClassMeta(type):
+    """Metaclass for constructing deprecated aliases of renamed classes.
+
+    Code implementation copied from https://stackoverflow.com/a/52087847
+    """
+
+    def __new__(cls, name, bases, classdict, *args, **kwargs):
+        alias = classdict.get("_DeprecatedClassMeta__alias")
+
+        if alias is not None:
+
+            def new(cls, *args, **kwargs):
+                alias = getattr(cls, "_DeprecatedClassMeta__alias")
+
+                if alias is not None:
+                    warn(
+                        f"{cls.__name__} has been renamed to {alias.__name__}, and the "
+                        f"alias will be removed in the future",
```

Comment:
```suggestion
                        f"alias will be removed in the Kedro 0.19.0",
```

Reply:
@merelcht Do we have another release (or few) before 0.19.0? With this change, we could also warn in 0.19.x at drop in 0.20.0 (or something), if there's any reason to give more time.

If there is no release before 0.19.0, then of course doesn't really help to have this.

Reply:
Yes we'll definitely have a couple more releases before `0.19.0`.